controller/api/public/stat_summary.go

```diff
@@ -92,50 +97,46 @@ func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest
 		}
 	}
 
-	statTables := make([]*pb.StatTable, 0)
-
-	var resourcesToQuery []string
-	if req.Selector.Resource.Type == k8s.All {
-		resourcesToQuery = k8s.StatAllResourceTypes
-	} else {
-		resourcesToQuery = []string{req.Selector.Resource.Type}
+	kind := req.GetSelector().GetResource().GetType()
+	if !isSupportedKind(kind) {
+		return nil, util.GRPCError(status.Errorf(codes.Unimplemented, "unimplemented resource type: %s", kind))
```

Comment:
Previously, we rely on [`k8sAPI.Objects()`](https://github.com/linkerd/linkerd2/blob/49d39e5a123ecf40e12d51f28d4185eb1bf7512f/controller/api/public/stat_summary.go#L162) to detect unsupported resource type. This change allows us to fail early, without creating any additional goroutines.